packages/opentelemetry-instrumentation/src/platform/browser/instrumentation.ts

```diff
@@ -16,20 +16,29 @@
 
 import { InstrumentationAbstract } from '../../instrumentation';
 import * as types from '../../types';
+import { InstrumentationBaseBrowser } from './types';
 
 /**
  * Base abstract class for instrumenting web plugins
  */
 export abstract class InstrumentationBase
   extends InstrumentationAbstract
-  implements types.Instrumentation {
+  implements InstrumentationBaseBrowser {
+  private _timer?: number;
+
   constructor(
     instrumentationName: string,
     instrumentationVersion: string,
     config: types.InstrumentationConfig = {}
   ) {
     super(instrumentationName, instrumentationVersion, config);
+    this._timer = window?.setTimeout(() => {
```

Comment:
if I understand the browser implemenation correct `loadInstrumentation()` could be called in base class once for all.

Reply:
it is exactly the situation we have now, calling this in "upper" class makes privates to be unreachable

Reply:
But it seems this is only a problem for node which does module patching

Reply:
check the code the load also enables the package, you can't enable this earlier because you will hit exactly the same problem - no privates, so this iw what it looks like. Either no privates or enforcing the `loadInstrumentation` in constructor to be called immediately. This will keeps the consistency too. If you have better way of solving such issue that you will have privates and you can call `loadInstrumentation` in "upper" class, I'm open for suggestion. So far I don't see anything better, if you know how to eat cookie and have cookie pls do tell I will be happy to make it :)

Reply:
My feeling is that the automatic/implict `enable()` is the problem. See also https://github.com/open-telemetry/opentelemetry-js/issues/2410#issuecomment-902421182

Reply:
this PR fixes the problem you mentioned too

Reply:
Why do we not just remove the `enable` option from the config and rely on calling `enable()` externally instead? Then in `enable` all constructors are guaranteed to be finished and there is no problem.